### DISCUSSION TITLE
System error: Unknown Error when setting up request to TabPy. null

### DISCUSSION THREAD

**Discussion by tb582** [tableau/TabPy#621] [Q&A]
Anyone familiar with that error ? seems to be captured in backendProcessManager.log but System error: Unknown Error when setting up request to TabPy. null is the only thing returned to prep :( 

`{"ts":"2023-10-17T12:32:19.972","sev":"info","k":"backend-manager","v":"Java path: /Applications/Tableau Prep Builder 2023.2.app/Contents/Plugins/jre/bin/java","pid":8411}
{"ts":"2023-10-17T12:32:19.977","sev":"info","k":"backend-manager","v":"spawned restApi with PID: 8417","pid":8411}
{"ts":"2023-10-17T12:32:57.184","sev":"info","k":"backend-manager","v":"New connection","pid":8411}
{"ts":"2023-10-17T12:32:57.188","sev":"info","k":"backend-manager","v":"Received HTTP request from backend","pid":8411}
{"ts":"2023-10-17T12:32:57.189","sev":"info","k":"backend-manager","v":"Got POST /backendConfig request","pid":8411}
{"ts":"2023-10-17T12:32:57.190","sev":"info","k":"backend-manager","v":"Got {\"port\":53272}","pid":8411}
{"ts":"2023-10-17T12:32:57.190","sev":"info","k":"backend-manager","v":"Successfully started backend","pid":8411}
{"ts":"2023-10-17T12:32:57.193","sev":"info","k":"backend-manager","v":"Destroying a client connection","pid":8411}
{"ts":"2023-10-17T12:32:57.194","sev":"info","k":"backend-manager","v":"Closing HTTP server","pid":8411}
{"ts":"2023-10-17T12:33:35.089","sev":"info","k":"rest-api-stdout","v":"WARNING: An illegal reflective access operation has occurred","pid":8411}
{"ts":"2023-10-17T12:40:22.309","sev":"info","k":"rest-api-stdout","v":"\nWARNING: Illegal reflective access by com.tableau.loom.utils.ConnectionUtils (jar:file:/Applications/Tableau%20Prep%20Builder%202023.2.app/Contents/Resources/app/loom-rest-api-1.0-SNAPSHOT.jar!/BOOT-INF/lib/loom-lang-20232.0.jar!/) to method sun.nio.fs.UnixPath.asByteArray()\nWARNING: Please consider reporting this to the maintainers of com.tableau.loom.utils.ConnectionUtils\nWARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations\nWARNING: All illegal access operations will be denied in a future release\njava.util.concurrent.ExecutionException: com.tableau.loom.lang.api.exceptions.LangSystemException: System error: Unknown Error when setting up request to TabPy. null","pid":8411}
{"ts":"2023-10-17T12:40:22.309","sev":"info","k":"rest-api-stdout","v":"\n\tat com.tableau.loom.runtime.FlowEvaluationJobCompletion.get(FlowEvaluationJobCompletion.java:89)","pid":8411}
{"ts":"2023-10-17T12:40:22.309","sev":"info","k":"rest-api-stdout","v":"\n\tat com.tableau.loom.runtime.FlowEvaluationJobCompletion.get(FlowEvaluationJobCompletion.java:35)\n\tat com.tableau.loom.execution.engine.FlowRunner.call(FlowRunner.java:50)\n\tat com.tableau.loom.execution.engine.FlowRunner.call(FlowRunner.java:36)\n\tat com.tableau.loom.orchestration.execution.CallableWrapperWithSecurityContext.call(CallableWrapperWithSecurityContext.java:44)\n\tat java.base/java.util.concurrent.FutureTask.run(Unknown Source)\n\tat com.tableau.loom.orchestration.execution.RunnableWrapperWithSecurityContext.run(RunnableWrapperWithSecurityContext.java:43)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.base/java.lang.Thread.run(Unknown Source)\nCaused by: com.tableau.loom.lang.api.exceptions.LangSystemException: System error: Unknown Error when setting up request to TabPy. null","pid":8411}
{"ts":"2023-10-17T12:40:22.309","sev":"info","k":"rest-api-stdout","v":"\n\tat com.tableau.loom.runtime.EvaluationInstance.waitForFlow(EvaluationInstance.java:288)\n\tat com.tableau.loom.runtime.FlowEvaluationJobCompletion.get(FlowEvaluationJobCompletion.java:73)\n\t... 9 more\nCaused by: com.tableau.loom.pythonSupport.PythonScriptException: Unknown Error when setting up request to TabPy. null\n\tat com.tableau.loom.pythonSupport.PythonExecutor.withHttpClient(PythonExecutor.java:140)\n\tat com.tableau.loom.pythonSupport.PythonExecutor.run(PythonExecutor.java:107)\n\tat com.tableau.loom.plugin.execution.PluginEntityRunner.execute(PluginEntityRunner.java:159)\n\tat com.tableau.loom.plugin.execution.PluginEntityRunner.selectIntoAsync(PluginEntityRunner.java:104)\n\tat com.tableau.loom.runtime.SelectIntoExecutor.call(SelectIntoExecutor.java:590)\n\tat com.tableau.loom.runtime.SelectIntoExecutor.call(SelectIntoExecutor.java:121)","pid":8411}
`

**Comment by mattdotgreenlake** [comment]
Consider the source of possible nulls
https://community.tableau.com/s/question/0D58b00009nPrUGCA0/i-cant-seem-to-get-past-system-error-unknown-error-when-setting-up-request-to-tabpy-null-when-including-a-python-script-in-my-prep-flow-using-tabpy

**Reply by tb582** [reply]
Sorry i should of clarified that I already ensured there’s n blank, null or NAN values in the df

**Reply by mattdotgreenlake** [reply]
Null is your only clue so far.  Its coming from somewhere. You'll need to follow the method of the community poster and start tracing/printing variables until you find what NULL is being passed.

**Reply by tb582** [reply]
I'm telling you there's no null values in the final DF. And the error is thrown by tableau prep shortly after the flow starts running yet the script still completes without issue.
